game/screenshots.rpy

```diff
@@ -0,0 +1,196 @@
+label screenshot_dialogue:
+    if _player_screenshot_in_progress:
+        # Don't take a screenshot if we're already going through the dialogue!
+        return
+    else:
+        $ _player_screenshot_in_progress = True
+
+    if persistent.jn_first_screenshot_taken is None:
+        # Set the date for the first ever screenshot, play the camera effects
+        $ persistent.jn_first_screenshot_taken = datetime.datetime.now()
+        call take_screenshot
+
+        n "H-huh?{w=0.2} What was that flash I just saw?"
+        n "Don't tell me...{w=0.3} was that a camera?!{w=0.2} There's a camera here?!"
+        n "..."
+        n "[player]...{w=0.3} d-did you do that...?"
+        menu:
+            "Yes, I did.":
+                n "O-oh!{w=0.2} Aha!{w=0.2} W-well,{w=0.1} at least you admit it."
+            "No, I didn't.":
+                n "Huh?{w=0.2} But then...{w=0.3} who...?"
+                n "..."
+            "I'm not sure.":
+                n "That's...{w=0.3} a little worrying..."
+                n "..."
+        n "Well, anyway.{w=0.1} The truth is,{w=0.1} I've never been very comfortable with having my picture taken without my permission."
+        n "I just...{w=0.3} really,{w=0.1} really don't like it."
+        n "So for the future,{w=0.1} could you please just let me know if you want to take pictures?"
+        n "I'd really appreciate it,{w=0.1} [player]."
+
+    # Positive screenshot route, as we have Natsuki's permission
+    elif player_screenshots_permission:
+        python:
+            # Update tracking and take the shot
+            persistent.jn_screenshot_good_shots_total += 1
+        n "Huh?{w=0.2} You're taking that picture now?"
+
+        if persistent.affinity >= 700:
+            n "Ahaha!{w=0.2} Sure!"
+            call take_screenshot
+        elif persistent.affinity < 700 and persistent.affinity > 300:
+            n "Well...{w=0.2} alright."
+            call take_screenshot
+        else:
+            n "...Fine.{w=0.1} Just be quick."
+            call take_screenshot
+
+        # Retract the permission Natsuki gave, as the picture has been taken
+        if persistent.affinity > 300:
+            n "Okaaay!{w=0.2} Just ask me again if you wanna take another,{w=0.1} alright?"
+        else:
+            n "All done?{w=0.2} Just ask me again if you wanna take another,{w=0.1} okay?"
+        $ player_screenshots_permission = False
+
+    # Too many bad screenshots in a row; Natsuki is upset
+    elif bad_screenshot_streak >= 3 and persistent.affinity < 700:
+        $ player_screenshots_blocked = True
+        call take_screenshot
+        n "Okay,{w=0.1} I think I've had enough!{w=0.2} I'm just gonna turn this off for now."
+        return
+
+    # Negative screenshot route; Natsuki is upset
+    elif not player_screenshots_blocked:
+        python:
+            # Update tracking and take the shot
+            persistent.jn_screenshot_bad_shots_total += 1
+            bad_screenshot_streak += 1
```

Comment:
Given this is a smaller section of pycode, we can use single-line py statements for a bit more cleanliness
```suggestion
        # Update tracking and take the shot
        $ persistent.jn_screenshot_bad_shots_total += 1
        $ bad_screenshot_streak += 1
```